File: wrappers/mvc/demos/Kendo.Mvc.Examples/Controllers/DataViz/Donut_Charts/Remote_DataController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web.Mvc;
using Kendo.Mvc.Examples.Models;


namespace Kendo.Mvc.Examples.Controllers
{
    public partial class Donut_ChartsController : Controller
    {
        public ActionResult Remote_Data()
        {
            return View();
        }

        private Dictionary<int, string> resolutionColors = new Dictionary<int, string>() { 
            {1,"#ccc"},
            {2,"#c00"}
        };

        public ActionResult _WorldScreenResolution()
        {

            var screenResolutions = ChartDataRepository.WorldScreenResolution();
            var viewModel = new List<ScreenResolutionRemoteDataViewModel>();

            for (var i = 0; i < screenResolutions.Count; i++)
            {
                var data = screenResolutions[i];
                var model = new ScreenResolutionRemoteDataViewModel(data);
                if (model.Year == "2005" && model.Resolution == "1024x768")
                {
                    model.Color = resolutionColors[2];
                }
                else if (model.Resolution == "Other")
                {
                    model.Color = resolutionColors[1];
                }

                viewModel.Add(model);
            }

            return Json(viewModel);
        }
    }
}

File: wrappers/java/spring-demos/src/main/java/com/kendoui/spring/controllers/donut/RemoteDataController.java
package com.kendoui.spring.controllers.donut;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

import com.kendoui.spring.models.DonutChartsRemoteDataViewModel;
import com.kendoui.spring.models.ChartDataRepository;
import com.kendoui.spring.models.ScreenResolution;

@Controller("dataviz-donut_charts_remote_data-controller")
@RequestMapping(value="/dataviz/donut-charts/")
public class RemoteDataController {
    private Map<Integer, String> resolutionColors = new HashMap<Integer, String>(){{
        put(1,"#ccc");
        put(2,"#c00");
    }};
    
    @RequestMapping(value = "/remote-data", method = RequestMethod.GET)
    public String index() {
        return "/dataviz/donut-charts/remote-data";
    }
    
    @RequestMapping(value = "/remote-data/read", method = RequestMethod.POST)
    public @ResponseBody List<DonutChartsRemoteDataViewModel> read() {
        
        
        List<ScreenResolution> screenResolutions = ChartDataRepository.WorldScreenResolution();
        List<DonutChartsRemoteDataViewModel> viewModel =  new ArrayList<DonutChartsRemoteDataViewModel>();
        
        for (int i = 0; i < screenResolutions.size(); i++)
        {
            ScreenResolution data = screenResolutions.get(i);
            DonutChartsRemoteDataViewModel item = new DonutChartsRemoteDataViewModel(data);
            if (item.getYear() == "2005" && item.getResolution() == "1024x768")
            {
                item.setColor(resolutionColors.get(2));
            } else if (item.getResolution() == "Other") {
                item.setColor(resolutionColors.get(1));
            }

            viewModel.add(item);
        }
        return viewModel;
    }
}
